File: problems/problems_55/Solution.java
package problems.problems_55;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public boolean canJump(int[] nums) {
        int maxDis = 0, n = nums.length;
        for (int i = 0; i < n; i++) {
            maxDis = Math.max(maxDis, i + nums[i]);
            if (maxDis >= n - 1) {
                return true;
            }
            if (i >= maxDis) {
                return false;
            }
        }
        return false;
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] nums = jsonArrayToIntArray(inputJsonValues[0]);
        return JSON.toJSON(canJump(nums));
    }
}

File: problems/problems_55/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.canJump(test_input)

    def canJump(self, nums: List[int]) -> bool:
        max_idx = 0
        for i, dis in enumerate(nums):
            max_idx = max(max_idx, i + dis)
            if max_idx >= len(nums) - 1:
                return True
            if i >= max_idx:
                return False
        return max_idx >= len(nums) - 1
